mlir/lib/IR/Builders.cpp

```diff
@@ -316,6 +316,15 @@ ArrayAttr Builder::getStrArrayAttr(ArrayRef<StringRef> values) {
   return getArrayAttr(attrs);
 }
 
+ArrayAttr
+Builder::getArrayOfI32ArrayAttr(ArrayRef<SmallVector<int32_t>> values) {
+  auto attrs =
+      llvm::map_to_vector<8>(values, [this](ArrayRef<int32_t> v) -> Attribute {
+        return getDenseI32ArrayAttr(v);
+      });
+  return getArrayAttr(attrs);
+}
```

Comment:
This isn't used?

Reply:
It isn't used currently. Previous this was used by the sharding option attribute logic. The PR description has also been modified to align this change.